server/public/model/property_group.go

```diff
@@ -0,0 +1,15 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package model
+
+type PropertyGroup struct {
+	ID   string
+	Name string
+}
+
+func (pg *PropertyGroup) PreSave() {
+	if pg.ID == "" {
+		pg.ID = NewId()
+	}
+}
```

Comment:
Do we need a `IsValid()` method that checks for the name?

Reply:
As the property group store accepts a `name string` instead of a group object, we're performing the `name == ""` check directly there